### PR TITLE
Add a mixin for comparing enums by index

### PR DESCRIPTION
**Pull request by elliette**
Closes https://github.com/flutter/devtools/issues/3999

Adds a new mixin for comparing enums by index, and uses the mixin for the `MediaSize` enum so that we can do comparisons like:

`if (screenWidth <= MediaSize.xs) ...`

### CODE REVIEW

File: packages/devtools_app/lib/src/primitives/enum_utils.dart [hunk 1]
```diff
@@ -32,3 +32,24 @@ class EnumUtils<T> {
 
   String? name(T enumEntry) => _reverseLookupTable[enumEntry];
 }
+
+mixin EnumIndexOrdering<T extends Enum> on Enum implements Comparable<T> {
+  @override
+  int compareTo(T other) => index - other.index;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/primitives/enum_utils.dart:38]: is index an int? `=> index.compareTo(other.index)` might be more clear

File: packages/devtools_app/lib/src/shared/status_line.dart [hunk 2]
```diff
@@ -67,22 +67,21 @@ class StatusLine extends StatelessWidget {
...
       ],
-      if (!isNarrow && pageStatus != null) ...[
+      if (screenWidth > MediaSize.xs && pageStatus != null) ...[
         pageStatus,
         const BulletSpacer(),
       ],
-      buildConnectionStatus(context, isExtraNarrow),
-      if (isEmbedded && !isExtraNarrow) ...[
+      buildConnectionStatus(context, screenWidth),
+      if (screenWidth > MediaSize.xxs && isEmbedded) ...[
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/status_line.dart:84]: nit: we could put `screenWidth > MediaSize.xxs` since it is used twice

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/lib/src/primitives/enum_utils.dart (hunk 1 above)
Comment (elliette) [reply packages/devtools_app/lib/src/primitives/enum_utils.dart]: Done!

File: packages/devtools_app/lib/src/shared/status_line.dart (hunk 2 above)
Comment (elliette) [reply packages/devtools_app/lib/src/shared/status_line.dart]: Good point, done!

**Pull request merged**